flow/connectors/postgres/validate.go

```diff
@@ -130,3 +137,35 @@ func (c *PostgresConnector) CheckReplicationConnectivity(ctx context.Context) er
 
 	return conn.Close(ctx)
 }
+
+// Check if all tables have either a primary key or REPLICA IDENTITY FULL
+func (c *PostgresConnector) CheckIfTablesAreMirrorable(
+	ctx context.Context,
+	tables []*utils.SchemaTable,
+) error {
+	badTables := make([]string, 0, len(tables))
+	for _, table := range tables {
+		var canMirror sql.NullBool
+		err := c.conn.QueryRow(ctx, `SELECT
+		(con.contype = 'p' OR t.relreplident in ('i', 'f')) AS can_mirror
```

Comment:
nit: replident being index doesn't seem to be enough, we need to check if the index also exists. 

`Records the old values of the columns covered by the named index, that must be unique, not partial, not deferrable, and include only columns marked NOT NULL. If this index is dropped, the behavior is the same as NOTHING.`

Reply:
can refer to existing `getReplicaIdentityIndexColumns` in Postgres connector for this